### PR TITLE
Cleanup more auto mapping names 

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Just a follow-up PR of #21903.

Rebase on main once #21911 (or a fix from your side) being merged will be OK.

### CODE REVIEW

File: utils/check_repo.py [hunk 1]
```diff
@@ -652,16 +652,20 @@ def check_all_auto_mapping_names_in_config_mapping_names():
     failures = []
 
     # `TOKENIZER_PROCESSOR_MAPPING_NAMES` and `AutoTokenizer` is special, and don't need to follow the rule.
-    mapping_to_check = {
+    mappings_to_check = {
```
Comment (ydshieh) [inline utils/check_repo.py:655]: better with `s`

File: utils/check_repo.py [hunk 2]
```diff
@@ -652,16 +652,20 @@ def check_all_auto_mapping_names_in_config_mapping_names():
...
 
-    for name, mapping in mapping_to_check.items():
+    # Each auto modeling files contains multiple mappings. Let's get them in a dynamic way.
+    for module_name in ["modeling_auto", "modeling_tf_auto", "modeling_flax_auto"]:
+        module = getattr(transformers.models.auto, module_name)
+        # all mappings in a single auto modeling file
+        mapping_names = [x for x in dir(module) if x.endswith("_MAPPING_NAMES")]
+        mappings_to_check.update({name: getattr(module, name) for name in mapping_names})
+
+    for name, mapping in mappings_to_check.items():
```
Comment (ydshieh) [inline utils/check_repo.py:668]: easier than I expected :-) Hope you are happy with this approach too.

Comment (sgugger) [reply utils/check_repo.py:668]: Clever!

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

File: utils/check_task_guides.py [hunk 3]
```diff
@@ -72,14 +72,24 @@ def _find_text_in_file(filename, start_prompt, end_prompt):
     "document_question_answering.mdx": transformers_module.models.auto.modeling_auto.MODEL_FOR_DOCUMENT_QUESTION_ANSWERING_MAPPING_NAMES,
 }
 
+# This list contains model types used in some task guides that are not in `CONFIG_MAPPING_NAMES` (therefore not in any
+# `MODEL_MAPPING_NAMES` or any `MODEL_FOR_XXX_MAPPING_NAMES`).
+SPECIAL_TASK_GUIDE_TO_MODEL_TYPES = {
+    "summarization.mdx": ("nllb",),
+    "translation.mdx": ("nllb",),
+}
+
```
Comment (ydshieh) [inline utils/check_task_guides.py:81]: When cleaning up the entries in mappings with tasks, like removing `nllb` from `MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING_NAMES`, the check `check_model_list_for_task` fails.

I think the best way is to:
  - still removing `nllb` from XXX__MAPPING_NAMES
  - add this `SPECIAL_TASK_GUIDE_TO_MODEL_TYPES`

WDYT?

File: utils/check_task_guides.py [hunk 4]
```diff
@@ -72,14 +72,24 @@ def _find_text_in_file(filename, start_prompt, end_prompt):
...
     Return the list of models supporting given task.
     """
-    config_maping_names = TASK_GUIDE_TO_MODELS[task_guide]
+    model_maping_names = TASK_GUIDE_TO_MODELS[task_guide]
+    special_model_types = SPECIAL_TASK_GUIDE_TO_MODEL_TYPES.get(task_guide, set())
     model_names = {
-        code: name for code, name in transformers_module.MODEL_NAMES_MAPPING.items() if code in config_maping_names
+        code: name
+        for code, name in transformers_module.MODEL_NAMES_MAPPING.items()
+        if (code in model_maping_names or code in special_model_types)
```
Comment (ydshieh) [inline utils/check_task_guides.py:92]: usage of the newly added `SPECIAL_TASK_GUIDE_TO_MODEL_TYPES`

File: utils/check_task_guides.py [hunk 5]
```diff
@@ -72,14 +72,24 @@ def _find_text_in_file(filename, start_prompt, end_prompt):
     "document_question_answering.mdx": transformers_module.models.auto.modeling_auto.MODEL_FOR_DOCUMENT_QUESTION_ANSWERING_MAPPING_NAMES,
 }
 
+# This list contains model types used in some task guides that are not in `CONFIG_MAPPING_NAMES` (therefore not in any
+# `MODEL_MAPPING_NAMES` or any `MODEL_FOR_XXX_MAPPING_NAMES`).
+SPECIAL_TASK_GUIDE_TO_MODEL_TYPES = {
+    "summarization.mdx": ("nllb",),
+    "translation.mdx": ("nllb",),
```
Comment (ydshieh) [inline utils/check_task_guides.py:79]: When cleaning up the entries in mappings with tasks, like removing `nllb` from `MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING_NAMES`, the check `check_model_list_for_task` fails.

I think the best way is to:
  - still removing `nllb` from XXX__MAPPING_NAMES
  - add this `SPECIAL_TASK_GUIDE_TO_MODEL_TYPES` and use it below

WDYT?

File: utils/check_task_guides.py (hunk 4 above)
Comment (ydshieh) [inline utils/check_task_guides.py:92]: usage of the newly added `SPECIAL_TASK_GUIDE_TO_MODEL_TYPES`

File: utils/check_task_guides.py (hunk 5 above)
Comment (ydshieh) [reply utils/check_task_guides.py:79]: And FYI: I also applied the same approach to another check in the same file `check_all_auto_object_names_being_defined`, in order to check more mappings there too.

It finds `EfficientNetBackbone` is not defined in `transformers` and I removed `("efficientnet", "EfficientNetBackbone")` from `MODEL_FOR_BACKBONE_MAPPING_NAMES`.

Comment (sgugger) [reply utils/check_task_guides.py:79]: This works for me!

**Review by sgugger — APPROVED** [review]

**Pull request merged**